slm_lab/agent/algorithm/sac.py

```diff
@@ -179,10 +180,10 @@ def try_update_per(self, q_preds, q_targets):
 
     def train_alpha(self, alpha_loss):
         '''Custom method to train the alpha variable'''
-        self.alpha_lr_scheduler.step(epoch=self.body.env.clock.frame)
         self.alpha_optim.zero_grad()
         alpha_loss.backward()
         self.alpha_optim.step()
+        self.alpha_lr_scheduler.step(epoch=self.body.env.clock.frame)
```

Comment:
this has to go back to the top, otherwise it will break the resume capability and also lag the clocks.
throughout the lab clock ticks first before things happen, same for resuming: clock gets restored, along with clock-dependent variables, then updates can occur.

Reply:
Good point, I hadn't accounted for that. Reverted.